modules/core/lib/Auth/UserPassBase.php

```diff
@@ -169,6 +169,22 @@ public function isRememberMeChecked() {
 	public function authenticate(&$state) {
 		assert('is_array($state)');
 
+
+		if (isset($state['core:auth:username']) && isset($state['core:auth:password'])) {
+			/* It looks like we are doing HTTP Basic authentication. */
+			$username = $state['core:auth:username'];
+			$password = $state['core:auth:password'];
+
+			if ($this->forcedUsername !== NULL && $this->forcedUsername !== $username) {
+				throw new SimpleSAML_Error_Error('WRONGUSERPASS');
+			}
+
+			$attributes = $this->login($username, $password);
```

Comment:
We should probably surround this with a `try { ... } catch` statement and log the error.